### ISSUE TITLE
Implement a homing algorithm using the limit switches

### ISSUE THREAD

**Issue by JackBuck** [JackBuck/Robo-Plot#72]
Some points to consider:
  - This should *probably* be done after #71 .
  - The `Axis` class could be modified to have an `override_limit_switch` property to allow the home. The back-off move should still be called after the home however.
  - The home position should be at 'minimum' x and y. That is, step as far as you can in x and y with the `Axis.forwards` property set to false.
  - Beware that the plotter will likely hit just one axis first, then the other.
  - Do we really want (0,0) at the limit switch? We may need an offset to allow for the paper to begin inside the limit switch.

**Comment by JackBuck** [comment]
Also merged into M_LimitSwitches now.

Just waiting for a real implementation of the limit switches!

**Comment by JackBuck** [comment]
Most of this is ready for review and in the pull request for M_LimitSwitches #91 .
However, the threaded homing is still awaiting testing on the plotter (and is not part of the referenced pull request)

**Comment by JackBuck** [comment]
Threaded homing successfully tested at the practical session.

**Comment by JackBuck** [comment]
Fully merged :)
